eclipse-theia/theia
# don't send plugin metadata over JSON-RPC

#### What it does


- [ ] fix #6228:
  - don't send plugin metadata over JSON-RPC, see why and how [here](https://github.com/eclipse-theia/theia/issues/6228#issuecomment-533765703)
  - don't send debugger contributions from the plugin host process to the frontend, but instead initially them statically from plugin metadata, [see](https://github.com/eclipse-theia/theia/issues/6228#issuecomment-533856477)
- [ ] update the changelog with a breaking change

#### How to test


- remove native extension from the browse example and rebuild:
  - textmate grammar
  - debug-nodejs
  - java
  - java-debug
- install 57 VS Code extensions in `plugins` folder:
https://drive.google.com/file/d/1tuOsorMVOKi_twL9oAuZxgrD7-VUwNsj/view?usp=sharing
  - could not come up with a command to pull it from the gdrive in the terminal, if someone knows how to, please share
- use system proxy to emulate slow connection, e.g. Charles proxy
  - you cannot do it in Chrome, since it cannot throttle web sockets
- measure how long does it takes to activate all plugins before and after
- observe whether you get yellow status bar and ws frames hangs for 20-30 seconds after and before
- play with unstable connections to confirm that plugins get reloaded/unloaded on the reconnection

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
